main.go

```diff
@@ -544,6 +544,14 @@ func (s *objectiveServer) List(ctx context.Context, req *connect.Request[objecti
 					}
 				}
 			}
+			if oi.Indicator.BoolGauge != nil {
+				for _, m := range oi.Indicator.BoolGauge.LabelMatchers {
+					if rm, replace := groupingMatchers[m.Name]; replace {
+						m.Type = rm.Type
+						m.Value = rm.Value
+					}
+				}
```

Comment:
The problem is that the label name with the simple `up` and group by `job` doesn't contain the label name and won't be replaced by the code below. I'm pretty certain the same problem exists with the other indicators but never surfaced. 
```suggestion
				for _, m := range oi.Indicator.BoolGauge.LabelMatchers {
					if rm, replace := groupingMatchers[m.Name]; replace {
						m.Type = rm.Type
						m.Value = rm.Value
						delete(groupingMatchers, m.Name)
					}
				}
				// Same groupings don't exist in the label matchers as e.g. regex matchers.
				// These grouping matchers we simply need to add to the list of matchers.
				if len(groupingMatchers) > 0 {
					for _, m := range groupingMatchers {
						oi.Indicator.BoolGauge.LabelMatchers = append(oi.Indicator.BoolGauge.LabelMatchers, m)
					}
				}
			}
```